### DISCUSSION TITLE
Updating some icons

### DISCUSSION THREAD

**Discussion by pgilfernandez** [friction2d/friction#279] [Ideas]
Hi!

After successfully compile Friction under macOS, I have been tinkering around and found out some icons don't fit... maybe it's just me (I'm designer) but I took my time to redesign some icons and test them... to me they are much better now.
Instead of just doing a PR I would like to share them with you here in case you don't really like them or just to discuss about it:

## Alignment icons
I feel the weight of actual icons is to light compared with the rest of the GUI, I updated them all:
![friction-align](https://github.com/user-attachments/assets/d5e4e12b-d4e6-4664-b951-58c45b8cc0c0)

## Main tools
Some icons come from old icons from Inkscape and the concepts they created. Now a days we know a better approach is to design icons with "what they really do" instead of trying to draw the "tools we use for creating"... I don't know if I'm explaining myself flat, so here is my proposal:
![friction-tools](https://github.com/user-attachments/assets/43722dca-3701-4aa5-b148-4459e3cfcd50)

The point mode icon drinks from Illustrator idea of pointers but, now a days it explains very easily what it does "select and edit path" (nodes)...

The text tool also changed from "T" to "A" as that character feels more natural and not as "technical" as the "T". It also helps visually to swap from basic shapes as circles and squares and feel it more like a "text tool"...

Well, what do you think?

Indeed, they have been designed with Inkscape
Cheers

**Comment by rodlie** [comment]
Sure, updating some of the icons has been on my todo for a while. My only requirement is to keep them close to the blender [icon](https://ui.blender.org/icons) "style", since we use many of them.

Icons have their own repo: https://github.com/friction2d/friction-icon-theme

**Reply by pgilfernandez** [reply]
I often use Blender so I'm used to their GUI (impressive) and icons but I didn't know about that small website with all the icons, cool resource.

I will pack everything into a PR so that you can test them and decide if use them or not.

Furthermore, if on your TODO are more icons, just let me know and I try designing them. I love designing icons and it's not my first time ([FreeCAD](https://forum.freecad.org/viewtopic.php?t=18417), [Olive](https://github.com/olive-editor/olive/pull/1882), [Synfig](https://forums.synfig.org/t/modern-iconset-light-and-dark/13560), ...)

**Reply by rodlie** [reply]
> I will pack everything into a PR so that you can test them and decide if use them or not.

Great, do PR's against https://github.com/friction2d/friction-icon-theme and put new icons in `hicolor/scalable/friction`. I will move/resize/rename files as needed. The SVG's are source files, we use a PNG theme, generated in dedicated branch (`build-1.0.0`) and included in Friction through a submodule.

Regarding the align icons, we also need a replacement for the "center" icon:
https://github.com/friction2d/friction-icon-theme/blob/main/hicolor/scalable/legacy/pivot-align-center.svg

> Furthermore, if on your TODO are more icons, just let me know and I try designing them. I love designing icons and it's not my first time

Getting rid of "legacy" icons from enve would be nice, there are also several new icons that we should add. Will need to review what we have and need.

**Reply by pgilfernandez** [reply]
Oh, late, hehehe.
Well I can force push and move them all into `friction` folder
Yes, pivot-align-center is also included

> Getting rid of "legacy" icons from enve would be nice, there are also several new icons that we should add. Will need to review what we have and need.

OK, let me know or open an Issue with the ones that need to be updated

**Reply by rodlie** [reply]
> Oh, late, hehehe. Well I can force push and move them all into friction folder

Not needed, the PR's are good. I still commit icons to the legacy folder myself.

**Reply by rodlie** [reply]
The legacy folder is a mix of icons, it's legacy icons in general I mean.

**Comment by pgilfernandez** [comment]
Another proposal:

![color-stroke-background](https://github.com/user-attachments/assets/2cda4f83-33cf-4715-b841-225ccf896eae)

**Reply by rodlie** [reply]
Looks good to me :+1:

**Comment by rodlie** [comment]
Thanks for the PR's, I will merge them later today.

**Comment by Dinesh0N** [comment]
Add more blender and Inkscape icons ✨
![inkscape_6jRhIvVjbb](https://github.com/user-attachments/assets/537379b3-3fdb-479a-99d4-b1958e3130f8)
